app/src/main/java/com/android/sample/ui/workout/ImportScreen.kt

```diff
@@ -0,0 +1,174 @@
+import com.android.sample.model.workout.WorkoutType
+import com.android.sample.model.workout.WorkoutViewModel
+import com.android.sample.ui.composables.TopBar
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.theme.BlueWorkoutCard
+import com.android.sample.ui.theme.ContrailOne
+import com.android.sample.ui.theme.Dimensions
+import com.android.sample.ui.theme.FontSizes
+import com.android.sample.ui.theme.LightBackground
+import com.android.sample.ui.theme.Line
+import com.android.sample.ui.theme.OpenSans
+import com.android.sample.ui.theme.Shape
+import com.android.sample.ui.theme.TitleBlue
+
+/**
+ * Composable function for the import screen.
+ *
+ * @param navigationActions Actions for navigating between screens.
+ * @param workoutViewModel The viewmodel of the workouts
+ */
+@Composable
+fun ImportScreen(
+    navigationActions: NavigationActions,
+    workoutViewModel: WorkoutViewModel<Workout>,
+    workoutType: WorkoutType
+) {
+
+  val title =
+      when (workoutType) {
+        WorkoutType.BODY_WEIGHT -> R.string.TitleTabBody
+        else -> R.string.TitleTabYoga
+      }
+  Scaffold(
+      topBar = { TopBar(navigationActions, R.string.Import) },
+      containerColor = LightBackground,
+      modifier = Modifier.testTag("ImportScreen")) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(horizontal = Dimensions.LargePadding, vertical = 24.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center // Center the content vertically
+            ) {
+              Text(
+                  text = stringResource(id = title),
+                  style =
+                      MaterialTheme.typography.labelLarge.copy(
+                          fontSize = FontSizes.MediumBigTitleFontSize),
+                  fontFamily = OpenSans,
+                  color = TitleBlue,
+                  modifier = Modifier.height(50.dp).testTag("ExerciseTypeTitle"))
+              Divider(
+                  color = Line,
+                  thickness = 0.5.dp,
+                  modifier =
+                      Modifier.padding(horizontal = 25.dp, vertical = 1.dp)
+                          .padding(bottom = 10.dp)
+                          .shadow(1.dp))
+              WorkoutList(navigationActions, workoutViewModel)
+            }
+      }
+}
+
+/**
+ * Composable function that the list of done workouts.
+ *
+ * @param navigationActions Actions for navigating between screens.
+ * @param workoutViewModel The viewmodel of the workouts
+ */
+@Composable
+fun WorkoutList(navigationActions: NavigationActions, workoutViewModel: WorkoutViewModel<Workout>) {
+  val doneWorkouts = workoutViewModel.doneWorkouts.collectAsState().value
+
+  LazyColumn(
```

Comment:
Maybe it would be good to add a fallback UI for cases where doneWorkouts is empty (e.g., "No workouts found.").

Reply:
Nice catch ! Done.